edb/schema/reflection/reader.py

```diff
@@ -0,0 +1,227 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2020-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import annotations
+from typing import *
+
+import collections
+import functools
+import json
+import uuid
+
+import immutables
+
+from edb.common import uuidgen
+
+from edb.schema import expr as s_expr
+from edb.schema import functions as s_func
+from edb.schema import name as s_name
+from edb.schema import objects as s_obj
+from edb.schema import operators as s_oper
+from edb.schema import schema as s_schema
+
+from . import structure as sr_struct
+
+
+def parse_into(
+    schema: s_schema.Schema,
+    data: Sequence[str],
+    schema_class_layout: Dict[Type[s_obj.Object], sr_struct.SchemaTypeLayout],
+) -> s_schema.Schema:
```

Comment:
Can you add a docstring? My assumption is that we're parsing & applying the introspection JSON here?